components/financial_assistance/app/domain/financial_assistance/operations/applications/pvc/create_pvc_request.rb

```diff
@@ -56,16 +56,76 @@ def create_non_esi_evidences(application)
             end
 
             Success(true)
+          rescue StandardError => e
+            pvc_logger.error("Failed to create non_esi_evidences for application with hbx_id #{application.hbx_id} due to #{e.inspect}")
+            Failure("Failed to create non_esi_evidences for application with hbx_id #{application.hbx_id}")
           end
 
-          def transform_application(application)
-            payload = ::FinancialAssistance::Operations::Applications::Transformers::ApplicationTo::Cv3Application.new.call(application)
-            AcaEntities::MagiMedicaid::Operations::InitializeApplication.new.call(payload.value!)
+          def transform_and_validate_application(application)
+            payload_entity = ::Operations::Fdsh::BuildAndValidateApplicationPayload.new.call(application)
+            record_application_failure(application, payload_entity.failure.messages) if payload_entity.failure?
+
+            if EnrollRegistry.feature_enabled?(:validate_and_record_publish_application_errors) && payload_entity.success?
+              all_applicants_valid = validate_applicants(payload_entity, application)
+              return all_applicants_valid.all?(true) ? payload_entity : Failure("Failed to transform application with hbx_id #{application.hbx_id} due to all applicants being invalid")
+            end
+
+            payload_entity
           rescue StandardError => e
             pvc_logger.error("Failed to transform application with hbx_id #{application.hbx_id} due to #{e.inspect}")
+            record_application_failure(application, "transformation failure") if payload_entity.failure?
```

Comment:
This is a feature specific requirement, For DC this feature is disabled. We should not record and show the new message if the feature is disabled.